src/rpc/bobtail.cpp

```diff
@@ -0,0 +1,245 @@
+// Copyright (c) 2020 The Bitcoin Unlimited developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include "amount.h"
+#include "blockrelay/netdeltablocks.h"
+#include "blockstorage/blockstorage.h"
+#include "bobtail/bobtail.h"
+#include "bobtail/dag.h"
+#include "bobtail/miner.h"
+#include "bobtail/validation.h"
+#include "chain.h"
+#include "chainparams.h"
+#include "consensus/consensus.h"
+#include "consensus/params.h"
+#include "consensus/validation.h"
+#include "core_io.h"
+#include "dstencode.h"
+#include "init.h"
+#include "main.h"
+#include "net.h"
+#include "parallel.h"
+#include "pow.h"
+#include "rpc/server.h"
+#include "txadmission.h"
+#include "txmempool.h"
+#include "ui_interface.h"
+#include "util.h"
+#include "utilstrencodings.h"
+#include "validationinterface.h"
+
+#include <cstdlib>
+#include <stdint.h>
+
+#include <boost/assign/list_of.hpp>
+#include <boost/shared_ptr.hpp>
+
+extern CBobtailDagSet bobtailDagSet;
+
+UniValue generateBobtailBlocks(boost::shared_ptr<CReserveScript> coinbaseScript,
+    int nGenerate,
+    uint64_t nMaxTries,
+    bool keepScript,
+    int weak_mode)
+{
+    static const int nInnerLoopCount = 0x10000;
+
+    unsigned int nExtraNonce = 0;
+    UniValue blockHashes(UniValue::VARR);
+
+    int numblocks = 0;
+
+    while (numblocks < nGenerate)
+    {
+        std::unique_ptr<CSubBlockTemplate> pblocktemplate;
+        {
+            TxAdmissionPause lock; // flush any tx waiting to enter the mempool
+            pblocktemplate = SubBlockAssembler(Params()).CreateNewSubBlock(coinbaseScript->reserveScript);
+        }
+        if (!pblocktemplate.get())
+            throw JSONRPCError(RPC_INTERNAL_ERROR, "Couldn't create new block");
+        LOG(WB, "Using delta block for RPC generate.\n");
+        CSubBlock *pblock = pblocktemplate->subblock.get();
+        {
+            // LOCK(cs_main);
+            IncrementExtraNonce(pblock, nExtraNonce);
+        }
+
+        // Generally look for weak PoW
+        while (nMaxTries > 0 && pblock->nNonce < nInnerLoopCount &&
+               !CheckProofOfWork(pblock->GetHash(), weakPOWfromPOW(pblock->nBits), Params().GetConsensus(), true))
```

Comment:
`weakPOWfromPOW` was the old way of checking weak PoW in deltablocks. It should probably be removed from bobtail.h.

I believe the line should actually look like this:
`while (nMaxTries > 0 && pblock->nNonce < nInnerLoopCount && !CheckSubBlockPoW(pblock->header, Params().GetConsensus(), BOBTAIL_K)`

Reply:
removed that function and fixed the rpc call